Armadillo/src/main/java/com/scribd/armadillo/encryption/SecureStorage.kt

```diff
@@ -71,48 +72,53 @@ internal class ArmadilloSecureStorage @Inject constructor(
     }
 
     override fun saveDrmDownload(context: Context, audioUrl: String, drmDownload: DrmDownload) {
+        if(secureDrmStorage == null){
+            throw DrmDownloadException(UnsupportedOperationException("This device cannot encrypt downloads"))
```

Comment:
What should the user experience be if this happens?

Reply:
Download failed.

Reply:
Is there anything they can do to fix it?

Reply:
No. If their device is in the failed mode they can't download.